networkx/drawing/nx_pylab.py

```diff
@@ -890,6 +891,207 @@ def _connectionstyle(posA, posB, *args, **kwargs):
+
+        See `matplotlib.patches.ArrowStyle` for more options.
+
+    connectionstyle : string (default="arc3")
+        For multiedges, connectionstyle is overriden by the method. For other edges,
+        pass the connectionstyle parameter to create curved arc of rounding
+        radius rad. For example, connectionstyle='arc3,rad=0.2'.
+        See `matplotlib.patches.ConnectionStyle` and
+        `matplotlib.patches.FancyArrowPatch` for more info.
+
+    arrowsize : int (default=10)
+        For directed graphs, choose the size of the arrow head's length and
+        width. See `matplotlib.patches.FancyArrowPatch` for attribute
+        `mutation_scale` for more info.
+
+    node_size : scalar or array (default=300)
+        Size of nodes. Though the nodes are not drawn with this function, the
+        node size is used in determining edge positioning.
+
+    nodelist : list, optional (default=G.nodes())
+       This provides the node order for the `node_size` array (if it is an array).
+
+    node_shape :  string (default='o')
+        The marker used for nodes, used in determining edge positioning.
+        Specification is as a `matplotlib.markers` marker, e.g. one of 'so^>v<dph8'.
+
+    label : None or string
+        Label for legend
+
+    min_source_margin : int (default=0)
+        The minimum margin (gap) at the begining of the edge at the source.
+
+    min_target_margin : int (default=0)
+        The minimum margin (gap) at the end of the edge at the target.
+
+    Example
+    -------
+    >>> G = nx.MultiGraph([(0, 1), (0, 1), (0, 2), (0, 2), (0, 2), (3, 0)])
+    >>> positions = {0: (0, 0), 1: (1, -2), 2: (2, 0), 3: (2, 2)}
+    >>> nx.draw_networkx_multigraph_edges(G, pos=positions)
+
+    Notes
+    -----
+    For directed graphs, arrows are drawn at the head end.  Arrows can be
+    turned off by passing an arrowstyle without
+    an arrow on the end.
+
+    Be sure to include `node_size` as a keyword argument; arrows are
+    drawn considering the size of nodes.
+
+    Self-loops are always drawn with `~matplotlib.patches.FancyArrowPatch`
+    regardless of the value of `arrows` or whether `G` is directed.
+    When ``arrows=False`` or ``arrows=None`` and `G` is undirected, the
+    FancyArrowPatches corresponding to the self-loops are not explicitly
+    returned. They should instead be accessed via the ``Axes.patches``
+    attribute (see examples)."""
+
+    if edgelist == None:
+        edgelist = G.edges()
+
+    if arrowstyle == None:
+        if G.is_directed():
+            arrowstyle = "-|>"
+        else:
+            arrowstyle = "-"
+
+    multiEdgeDict = {}
+    for edge in edgelist:
+        if edge not in multiEdgeDict:
+            multiEdgeDict[edge] = 1
+        else:
+            multiEdgeDict[edge] += 1
+
+    for edge in multiEdgeDict:
+        if multiEdgeDict[edge] == 1 and list(G.edges()).count((edge[1], edge[0])) == 0:
+            nx.draw_networkx_edges(
```

Comment:
To reduce the number of graphical edge containers (and calls to draw_*_edges) maybe it's best to collect all the singular edges in one list/dict and all the multiedge 2-tuples in another list/dict.  Then draw the singular edges with one connectionstyle function call and the multiedges with the arc you compute. I'm not sure that connectionstyle can vary across the edges in a FancyArrowPatch. If not, then you'll have to call the draw function for each multtiedge as you do now. But there would be only 1 LineCollection for the other non-multi edges.